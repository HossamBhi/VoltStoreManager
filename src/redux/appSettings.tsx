import {createSlice} from '@reduxjs/toolkit';
import {_getAppLanguages, _getAppThemes} from '../utils/appDB';

const initialState = {
  activeTheme: _getAppThemes()[0],
  activeLanguage: _getAppLanguages()[0],
};

const appSettings = createSlice({
  name: 'appSettings',
  initialState,
  reducers: {
    changeThemeAction: (state, {payload}) => {
      state.activeTheme = payload;
    },
    changeLanguageAction: (state, {payload}) => {
      state.activeLanguage = payload;
    },
  },
});

export const {changeThemeAction, changeLanguageAction} = appSettings.actions;
export default appSettings.reducer;
