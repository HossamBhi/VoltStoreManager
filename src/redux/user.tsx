import {createSlice} from '@reduxjs/toolkit';
const dummy_user = {
  avatar_urls: {'24': '', '48': '', '96': ''},
  description: '',
  id: 98,
  is_super_admin: false,
  link: '',
  name: '',
  password: '',
  slug: '',
  url: '',
  username: '',
};
const initialState: {
  logedUser: null | typeof dummy_user;
} = {
  logedUser: null,
};

const user = createSlice({
  name: 'users',
  initialState,
  reducers: {
    setLogedUserAction: (state, {payload}) => {
      state.logedUser = payload;
    },
    logOutAction: state => {
      state.logedUser = null;
    },
  },
});

export const {setLogedUserAction, logOutAction} = user.actions;
export default user.reducer;
