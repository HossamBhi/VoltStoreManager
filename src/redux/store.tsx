import {configureStore} from '@reduxjs/toolkit';
import {persistReducer} from 'redux-persist';
import AsyncStorage from '@react-native-async-storage/async-storage';
import {combineReducers} from 'redux';
import user from './user';
import appSettings from './appSettings';

const persistConfig = {
  key: 'root',
  version: 1,
  storage: AsyncStorage,
};

const rootReducer = combineReducers({user, appSettings});
const persistedReducer = persistReducer(persistConfig, rootReducer);
export type RootState = ReturnType<typeof rootReducer>;

export const store = configureStore({
  reducer: persistedReducer,
  middleware: getDefaultMiddleware =>
    getDefaultMiddleware({serializableCheck: false}),
});
