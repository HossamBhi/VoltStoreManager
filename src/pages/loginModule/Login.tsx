/* eslint-disable react-native/no-inline-styles */
import React, {useEffect, useState} from 'react';
import {View, ScrollView, useWindowDimensions, StyleSheet} from 'react-native';
import {t} from 'i18next';
import {useTheme} from 'react-native-paper';
import {Formik} from 'formik';
import {useDispatch, useSelector} from 'react-redux';
import * as Yup from 'yup';

import PageHeader from '../../components/pageHeader';
import BlockBtn from '../../components/common/BlockBtn';
import CustomeIcon from '../../components/common/CustomeIcon';
import CustomeTextInputWithHeader from '../../components/common/CustomeTextInputWithHeader';
import {ReactNavigationProps} from '../../navigation/types';
import CustomeImage from '../../components/common/CustomeImage';
import {getUserDataAPI} from '../../utils/APIs';
import LoginErrorMessage from '../../components/loginModule/LoginErrorMessage';
import {setLogedUserAction} from '../../redux/user';
import AppLoader from '../../components/common/AppLoader';
import {RootState} from '../../redux/store';

interface LoginFields {
  username: string;
  password: string;
}

const Login = ({navigation, route}: ReactNavigationProps) => {
  const {height} = useWindowDimensions();
  const dispatch = useDispatch();
  const {logedUser} = useSelector((state: RootState) => state.user);
  const {colors} = useTheme();
  const [loader, setLoaderStatus] = useState(false);
  const [validationError, setValidationError] = useState<string | null>(null);
  // const [values, setValues] = useState<LoginFields>({
  //   username: 'alsaaidmarket3@gmail.com',
  //   password: 'alsaaid#market3',
  // });

  const [values, setValues] = useState<LoginFields>({
    username: '',
    password: '',
    // username: 'badrgreenmarket@gmail.com',
    // password: 'badr@greenmarket2',
  });

  useEffect(() => {
    if (logedUser) {
      navigation.replace('BottomTabs');
    } else {
      setLoaderStatus(false);
    }
  }, [navigation, logedUser, route]);

  const createAccountSchema = Yup.object().shape({
    username: Yup.string()
      .trim()
      .required(t('Required') + ''),
    password: Yup.string().required(t('Required') + ''),
  });

  // const loginResultHandler = (result: any) => {
  //   // console.log('Login result: ', result);
  //   if (!result.error) {
  //     getUserDataAPI(values).then(data => {
  //       if (!data.error) {
  //         dispatch(setLogedUserAction({...data, ...values}));
  //         navigation.replace('BottomTabs');
  //       } else if (result.message) {
  //         setValidationError(result.message);
  //       } else {
  //         setValidationError(result.error);
  //       }
  //       setLoaderStatus(false);
  //     });
  //   } else {
  //     if (result.message) {
  //       setValidationError(result.message);
  //     } else {
  //       setValidationError(result.error);
  //     }
  //     setLoaderStatus(false);
  //   }
  // };

  const handleOnSubmit = (fieldsVal: LoginFields) => {
    setValues(fieldsVal);
    setLoaderStatus(true);
    // postLoginAPI(fieldsVal).then(loginResultHandler);
    getUserDataAPI(fieldsVal).then(({data}) => {
      // console.log({data: data.name});
      if (!data.error) {
        dispatch(setLogedUserAction({...data, ...fieldsVal}));
        navigation.replace('BottomTabs');
      } else if (data.message) {
        setValidationError(data.message);
        setLoaderStatus(false);
      }
    });
  };

  if (loader) {
    return <AppLoader />;
  }
  return (
    <ScrollView
      showsVerticalScrollIndicator={false}
      style={{
        flex: 1,
        backgroundColor: colors.background,
        paddingTop: 30,
      }}>
      <View style={{minHeight: height - 250}}>
        <PageHeader
          showCart={false}
          showSearchFilter={false}
          showMenu={false}
          showLogo={false}
          customeCenterPart={
            <CustomeImage
              source={require('../../assets/volt.png')}
              style={{flex: 1, width: 120, height: 90}}
              resizeMode="center"
            />
          }
          showClose={false}
          style={styles.pageHeader}
        />
        {validationError && <LoginErrorMessage message={validationError} />}
        {/* <IntroPage title={t('loginIntroTitle')} style={{paddingBottom: 70}} /> */}
        <View style={styles.formStyle}>
          <Formik
            initialValues={values}
            onSubmit={handleOnSubmit}
            validationSchema={createAccountSchema}>
            {({
              handleChange,
              handleBlur,
              handleSubmit,
              values: formValues,
              errors,
            }) => (
              <View>
                <CustomeTextInputWithHeader
                  placeholder={t('usernameOrEmail') + ''}
                  onChangeText={handleChange('username')}
                  value={formValues.username}
                  onBlur={handleBlur('username')}
                  title={t('usernameOrEmail')}
                  containerStyle={[errors.username ? styles.error : {}]}
                  returnKeyType="next"
                  keyBoardType="email-address"
                  leftPart={
                    <CustomeIcon
                      name="mail"
                      size={20}
                      color={colors.onSurfaceVariant}
                    />
                  }
                />
                <CustomeTextInputWithHeader
                  placeholder={t('password') + ''}
                  onChangeText={handleChange('password')}
                  value={formValues.password}
                  onBlur={handleBlur('password')}
                  title={t('password')}
                  containerStyle={[errors.password ? styles.error : {}]}
                  secureTextEntry={formValues.password.length > 0}
                  leftPart={
                    <CustomeIcon
                      name="lock"
                      size={20}
                      color={colors.onSurfaceVariant}
                    />
                  }
                />

                <BlockBtn
                  text={t('login')}
                  onPress={handleSubmit}
                  style={{marginTop: 30}}
                />
              </View>
            )}
          </Formik>
        </View>
      </View>
    </ScrollView>
  );
};
const styles = StyleSheet.create({
  pageHeader: {paddingLeft: 0, marginHorizontal: 16, paddingBottom: 70},
  confirmBtn: {marginTop: 40},
  error: {borderColor: '#DC484B', borderWidth: 1},
  formStyle: {paddingHorizontal: 16},
});

export default Login;
