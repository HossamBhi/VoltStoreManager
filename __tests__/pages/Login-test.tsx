import React from 'react';
import {create} from 'react-test-renderer';
import Login from '../../src/pages/loginModule/Login';
import {Provider} from 'react-redux';
import {store} from '../../src/redux/store';
import {DefaultTheme, Provider as PaperProvider} from 'react-native-paper';

const navigation: any = {navigate: jest.fn()};
const route: any = {};

const tree = create(
  <Provider store={store}>
    <PaperProvider theme={DefaultTheme}>
      <Login {...{navigation, route}} />
    </PaperProvider>
  </Provider>,
);
test('Render Login Screen', () => {
  expect(tree).toMatchSnapshot();
});

/** 
 * TODO
 * invalid username
 * invalid password
 * check if user logedin 
*/
