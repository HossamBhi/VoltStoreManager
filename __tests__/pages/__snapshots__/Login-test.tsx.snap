// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Login Screen 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    collapsable={false}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <RCTScrollView
      showsVerticalScrollIndicator={false}
      style={
        Object {
          "backgroundColor": "rgba(255, 251, 254, 1)",
          "flex": 1,
          "paddingTop": 30,
        }
      }
    >
      <View>
        <View
          style={
            Object {
              "minHeight": 1084,
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "paddingHorizontal": 16,
                  "paddingVertical": 22,
                },
                Object {
                  "marginHorizontal": 16,
                  "paddingBottom": 70,
                  "paddingLeft": 0,
                },
              ]
            }
          >
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "justifyContent": "space-between",
                    "paddingBottom": 22,
                  },
                  undefined,
                ]
              }
            >
              <Image
                onError={[Function]}
                resizeMode="center"
                source={
                  Object {
                    "testUri": "../../../src/assets/volt.png",
                  }
                }
                style={
                  Array [
                    Object {
                      "minHeight": 30,
                      "minWidth": 30,
                    },
                    Object {
                      "flex": 1,
                      "height": 90,
                      "width": 120,
                    },
                  ]
                }
                testID="image"
              />
            </View>
          </View>
          <View
            style={
              Object {
                "paddingHorizontal": 16,
              }
            }
          >
            <View>
              <View>
                <Text
                  style={
                    Array [
                      Object {
                        "color": undefined,
                      },
                      Object {
                        "lineHeight": 11,
                        "paddingBottom": 4,
                        "paddingTop": 20,
                      },
                    ]
                  }
                />
                <View
                  accessibilityState={
                    Object {
                      "disabled": false,
                    }
                  }
                  accessible={true}
                  collapsable={false}
                  focusable={true}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Array [
                      Object {
                        "borderRadius": 10,
                      },
                      Array [
                        Object {
                          "alignItems": "center",
                          "flex": 1,
                          "flexDirection": "row",
                        },
                        Array [
                          Object {
                            "borderRadius": 10,
                            "paddingHorizontal": 15,
                            "paddingVertical": 8,
                          },
                          Object {
                            "backgroundColor": "rgba(231, 224, 236, 1)",
                          },
                          Array [
                            Object {},
                          ],
                        ],
                      ],
                      undefined,
                    ]
                  }
                >
                  <Text
                    allowFontScaling={false}
                    selectable={false}
                    style={
                      Array [
                        Object {
                          "color": "rgba(73, 69, 79, 1)",
                          "fontSize": 20,
                        },
                        undefined,
                        Object {
                          "fontFamily": "Feather",
                          "fontStyle": "normal",
                          "fontWeight": "normal",
                        },
                        Object {},
                      ]
                    }
                  >
                    
                  </Text>
                  <TextInput
                    keyBoardType="email-address"
                    leftPart={
                      <_default
                        color="rgba(73, 69, 79, 1)"
                        name="mail"
                        size={20}
                      />
                    }
                    onBlur={[Function]}
                    onChangeText={[Function]}
                    placeholder="undefined"
                    placeholderTextColor="rgba(73, 69, 79, 1)"
                    returnKeyType="next"
                    style={
                      Array [
                        Object {
                          "color": undefined,
                        },
                        Object {
                          "flex": 1,
                          "fontSize": 14,
                          "paddingHorizontal": 10,
                        },
                        undefined,
                      ]
                    }
                    value=""
                  />
                </View>
              </View>
              <View>
                <Text
                  style={
                    Array [
                      Object {
                        "color": undefined,
                      },
                      Object {
                        "lineHeight": 11,
                        "paddingBottom": 4,
                        "paddingTop": 20,
                      },
                    ]
                  }
                />
                <View
                  accessibilityState={
                    Object {
                      "disabled": false,
                    }
                  }
                  accessible={true}
                  collapsable={false}
                  focusable={true}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Array [
                      Object {
                        "borderRadius": 10,
                      },
                      Array [
                        Object {
                          "alignItems": "center",
                          "flex": 1,
                          "flexDirection": "row",
                        },
                        Array [
                          Object {
                            "borderRadius": 10,
                            "paddingHorizontal": 15,
                            "paddingVertical": 8,
                          },
                          Object {
                            "backgroundColor": "rgba(231, 224, 236, 1)",
                          },
                          Array [
                            Object {},
                          ],
                        ],
                      ],
                      undefined,
                    ]
                  }
                >
                  <Text
                    allowFontScaling={false}
                    selectable={false}
                    style={
                      Array [
                        Object {
                          "color": "rgba(73, 69, 79, 1)",
                          "fontSize": 20,
                        },
                        undefined,
                        Object {
                          "fontFamily": "Feather",
                          "fontStyle": "normal",
                          "fontWeight": "normal",
                        },
                        Object {},
                      ]
                    }
                  >
                    
                  </Text>
                  <TextInput
                    leftPart={
                      <_default
                        color="rgba(73, 69, 79, 1)"
                        name="lock"
                        size={20}
                      />
                    }
                    onBlur={[Function]}
                    onChangeText={[Function]}
                    placeholder="undefined"
                    placeholderTextColor="rgba(73, 69, 79, 1)"
                    secureTextEntry={false}
                    style={
                      Array [
                        Object {
                          "color": undefined,
                        },
                        Object {
                          "flex": 1,
                          "fontSize": 14,
                          "paddingHorizontal": 10,
                        },
                        undefined,
                      ]
                    }
                    value=""
                  />
                </View>
              </View>
              <View
                accessibilityState={
                  Object {
                    "disabled": false,
                  }
                }
                accessible={true}
                collapsable={false}
                focusable={true}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Array [
                    Object {
                      "borderRadius": 10,
                    },
                    Array [
                      Object {
                        "borderRadius": 10,
                        "paddingVertical": 17,
                      },
                      Object {
                        "backgroundColor": "rgba(103, 80, 164, 1)",
                      },
                      Object {
                        "marginTop": 30,
                      },
                    ],
                    undefined,
                  ]
                }
              />
            </View>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>
</RNCSafeAreaProvider>
`;
